quality-assurance/core.md

```diff
@@ -0,0 +1,193 @@
+assurance standards in order to consistently deliver a high quality software project.
+
+There are different parts/processes to quality assurance as a whole, and these are not limited to just manual testing and
+automated testing.
+
+*   [What makes a good web application?](#what-makes-a-good-web-application)
+    *   [Blackbox](#blackbox)  
+    *   [Whitebox](#whitebox)  
+*   [Processes](#processes)
+    *   [Automated Testing](#automated-testing)
+    *   [Manual Testing](#manual-testing)
+
+## What makes a good web application?
+The following is a checklist of things to think about both during development of a product and during the quality 
+assurance phase. They should be considered guidelines, and not a strict set of requirements, as not all points will be applicable or appropriate for all applications.
+
+### Blackbox
+**General**
+* All UI scales well and looks good on small form factor devices
+* All errors are handled and UI reacts appropriately
+* Errors are clear in meaning &amp; reflect what happened
+* Activity indicators are used when loading data asynchronously
+* Empty information is handled correctly in UI (optional fields etc)
+* Long running operations give a indication of progress if possible
+* Breadcrumbs are used where appropriate
+
+**Navbar**
+* UI reflects the current active page
+* An icon that is appropriate for the page (if applicable)
+
+**Forms**
+* Invalid inputs correctly trigger validation errors
+* All form inputs have associated labels
+* Appropriate validation for different inputs (phone number, postcode)
+* The forms are not too long &amp; and are grouped logically
+* The user is alerted of the form lifecycle (success, error)
+* When modals are used in the context of updates, ensure inputs are correctly pre populated.
+* Autofill is disabled on inputs where appropriate
+* Inputs have correct type property (emails have email type, phone numbers have tel type)
+
+**Modals**
+* Modals are animated on both display and dismiss
+* Modals are used where appropriate over pages
+
+**Formatting**
+* Dates are formatted correctly for the appropriate audience.
+* Monetary values are formatted correctly for the appropriate audience
+* Phone numbers are formatted correctly for the appropriate audience
+
+**Pagination**
+* UI correctly indicates current page
+* UI shows at least 2 page tabs ahead and below the current page
+
+**Login &amp; Auth**
+* Users can only access pages they are permitted access to.
+* Authentication when they are logged in
+* Login, registration (if applicable) and logout works as expected.
+* Reset password works as expected
+
+**Registration**
+* Email addresses are verified &amp; unique (if applicable)
+* Passwords meet minimum standards (min 8 chars)
+* Passwords should not have maximum length restrictions (allow at least 255 characters).
+
+**Emails**
+* Emails look good on small form factor devices
```

Comment:
Might be worth mentioning ensuring the HTML and CSS in the email works on the popular clients if custom HTML and CSS is being written. There are good resources such as [this guide](https://www.campaignmonitor.com/css/) which lets you search by email clients.